src/components/DeckMechanisms/index.js

```diff
@@ -351,6 +364,43 @@ export default class DeckMechanisms extends React.Component {
       deck: this.getIncreasedDeckWeight({ state, reset }),
     }))
 
+  resolveManaRNG = state => {
+    switch (this.state.RNG) {
+      case 'UNFRIENDLY': {
+        state.specifics.activeFrozenCores = 0
+        state.specifics.activeDawnsparks = 0
+        break
+      }
+
+      case 'REGULAR': {
+        const { activeFrozenCores, activeDawnsparks } = state.specifics
+
+        state.specifics.activeFrozenCores = getBinomialRandomVariableResult(
+          activeFrozenCores,
+          FROZEN_CORE_STAYS
+        )
+        state.specifics.activeDawnsparks = getBinomialRandomVariableResult(
+          activeDawnsparks,
+          DAWNSPARKS_STAYS
+        )
+
+        state.mana += state.specifics.activeFrozenCores * 3
+        state.mana +=
+          getBinomialRandomVariableResult(
+            state.specifics.activeDawnsparks,
+            DAWNSPARKS_HITS
+          ) * 4
```

Comment:
How comes you are recomputing it a second time? It seems you did it once on L382 already.

Reply:
Once to remove Dawnsparks that were killed by the opponent, once to calculate how many of those who survived actually hit an enemy unit and give mana

Reply:
Ah yes, of course!